src/main/java/io/github/oliviercailloux/y2018/apartments/utils/JsonConvert.java

```diff
@@ -78,27 +82,42 @@ public static void apartmentToJson(Apartment a, Path jsonPath) throws IOExceptio
 
 	/**
 	 * Gets the address field from an Address JSON.
+	 * 
+	 * We have to make sure that we have everything at every step <br>
+	 * It is for this reason that the code is filled with check
+	 * <p>
+	 * The contract for this API is returned either <code>"features": []<code> if
+	 * the address cannot be found or in shape <code>{"features": ["properties":
+	 * {"label": "AdressHere"}]}<code>
+	 * </p>
 	 *
-	 * @param jsonString <i>String</i> the Address into JSON format
-	 * @return <i>String</i> the address field
+	 * @param jsonString {@link String} the Address into JSON format
+	 * @return {@link String} the address field
+	 * 
+	 * @throws IllegalArgumentException in the case where the deserialization of the
+	 *                                  JSON encounters a problem
 	 */
-	public static String getAddressFromJson(String jsonString) {
-		Jsonb jsonb = JsonbBuilder.create();
-
-		final LinkedHashMap<?, ?> result = jsonb.fromJson(jsonString, LinkedHashMap.class);
-		LOGGER.info("Get address");
-
-		checkArgument(!result.get("address").toString().isEmpty() && result.containsKey("address"),
-				"There is no field adress in the JSON file.");
-		return result.get("address").toString();
+	public static String getAddressFromJson(String jsonString) throws IllegalArgumentException {
```

Comment:
IllegalArgumentException est définie comme une runtime, il faudrait éviter de la déclarer.
Serait-il possible de créer une exception dédiée ?
Dans le cas ou c'est une illegalArgument pour une autre raison, elle pourrait être catch par erreur

Reply:
Fix in f928f39f367274a146722e569fb63e3c5a84b617